core/src/com/fourinachamber/fortyfive/screen/general/AdvancedTextWidget.kt

```diff
@@ -7,78 +7,126 @@ import com.badlogic.gdx.graphics.g2d.GlyphLayout
 import com.badlogic.gdx.math.Affine2
 import com.badlogic.gdx.math.Matrix4
 import com.badlogic.gdx.scenes.scene2d.Actor
+import com.badlogic.gdx.scenes.scene2d.ui.WidgetGroup
+import com.badlogic.gdx.scenes.scene2d.utils.Layout
 import com.badlogic.gdx.utils.TimeUtils
 import com.fourinachamber.fortyfive.screen.ResourceHandle
-import com.fourinachamber.fortyfive.screen.ResourceManager
-import com.fourinachamber.fortyfive.screen.general.styles.StyleCondition
-import com.fourinachamber.fortyfive.screen.general.styles.StyleInstruction
-import com.fourinachamber.fortyfive.screen.general.styles.StyleManager
-import com.fourinachamber.fortyfive.screen.general.styles.StyledActor
+import com.fourinachamber.fortyfive.screen.general.styles.*
+import com.fourinachamber.fortyfive.utils.AdvancedTextParser
 import com.fourinachamber.fortyfive.utils.TemplateString
-import io.github.orioncraftmc.meditate.YogaNode
+import com.fourinachamber.fortyfive.utils.splitAt
+import com.fourinachamber.fortyfive.utils.zip
 import io.github.orioncraftmc.meditate.YogaValue
 import io.github.orioncraftmc.meditate.enums.YogaUnit
-import onj.value.OnjArray
 import onj.value.OnjNamedObject
 import onj.value.OnjObject
 import kotlin.math.sin
 
 open class AdvancedTextWidget(
-    advancedText: AdvancedText,
-    screen: OnjScreen
-) : CustomFlexBox(screen) {
+    private val defaults: OnjObject,
+    val screen: OnjScreen,
+) : WidgetGroup(), ZIndexActor, HoverStateActor, StyledActor {
 
-    private var nodesOfCurrentText: List<YogaNode> = listOf()
+    override var fixedZIndex: Int = -1
```

Comment:
i am not sure if i remember that correctly, but somehow -1 was the same as 0 (ik it shoudn't be, but thats in my memories)